src/libs/error.ts

```diff
@@ -13,18 +15,29 @@ export const reportError = async (title, obj) => {
   notify('error', title, { detail: await (obj instanceof Response ? obj.text() : obj) })
 }
 
+export type ErrorCallback = (error: unknown) => void | Promise<void>
+const withErrorHandlingFn = <R, F extends AnyPromiseFn>(
```

Comment:
breaking the function out from the currying step below is key to being able to properly type the function and leverage earlier work to let these types flow well through safeCurry and optionally withHandlers utility on the consuming side.